### PR TITLE
Add excludeDeprecated option to client codegen

### LINKED ISSUE #645: Add an option to codegen for including deprecated fields
Currently they are included, but we might want to omit them.

### PR DESCRIPTION
**Pull request by jeejeeone**
Add excludeDeprecated option to client codegen, when enabled excludes deprecated fields and enum values. Resolves https://github.com/ghostdogpr/caliban/issues/645

Note: not quite sure about codegen-sbt part 🔢

### CODE REVIEW

**Comment by jeejeeone** [comment]
Just realised there is https://github.com/ghostdogpr/caliban/issues/1253 which relates to this since you can have enum with only deprecated fields + same applies for type. Perhaps it's enough to generate compiling code in these cases?

File: vuepress/docs/docs/client-codegen.md [hunk 1]
```diff
@@ -83,13 +83,14 @@ The settings available on the `cs` (`CalibanSettings`) builder are:
 - `enableFmt`: Enable code formatting with scalafmt (default: `true`).
 - `extensibleEnums`: Generate a fallback case class for unknown enum values (default: `false`).
 - `headers` (only defined for `url` settings): Supply extra headers when fetching the schema from a URL.
+- `excludeDeprecated`: Exclude fields and enum values with deprecated directive (default: `false`).
```
Comment (ghostdogpr) [inline vuepress/docs/docs/client-codegen.md]: Thanks for updating the docs! Do you mind doing it in a separate PR, because docs are published right away and it would be better to update them after this is released 🙏

**Review by ghostdogpr — COMMENTED** [review]
LGTM!

File: vuepress/docs/docs/client-codegen.md (hunk 1 above)
Comment (jeejeeone) [reply vuepress/docs/docs/client-codegen.md]: No problem 👍 Probably on thu-sunday this week.

Comment (jeejeeone) [reply vuepress/docs/docs/client-codegen.md]: PR for docs https://github.com/ghostdogpr/caliban/pull/2171

**Comment by ghostdogpr** [comment]
Thanks!
